### PR TITLE
add lookup set

### PR DESCRIPTION
**Pull request by ad-tra**

### CODE REVIEW

File: db/src/lib.rs [hunk 1]
```diff
@@ -15,6 +16,7 @@ pub mod errors;
 pub mod list;
 pub mod logger;
 pub mod lookup;
+pub mod lookup_list;
```
Comment (Parth) [inline db/src/lib.rs]:
```suggestion
pub mod lookup_set;
```

File: db/src/lookup_list.rs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        }
+
+        Ok(repr)
+    }
+}
+
+impl<K, V> LookupSet<K, V>
+where
+    K: Hash + Eq + Serialize + DeserializeOwned,
+    V: Serialize + DeserializeOwned + Eq + Hash,
```
Comment (Parth) [inline db/src/lookup_list.rs]:
```suggestion
    K: Hash + Eq + Serialize,
    V: Serialize + Eq + Hash,
```

p sure you don't need these here (is causing some problems for me in core)

File: db/tests/lookup_list_tests.rs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    table1: LookupSet<u8, String>,
+    table2: LookupSet<u8, String>,
+    table3: LookupSet<u8, String>,
+    table4: LookupSet<u8, String>,
+    table5: LookupSet<u8, String>,
+}
+
+#[test]
+fn test() {
+    let dir = "/tmp/a/";
```
Comment (Parth) [inline db/tests/lookup_list_tests.rs]: you don't want this to collide with other tests, you could generate a random number or you can just see what the last letter used so far is and increment it

**Review by Parth — CHANGES_REQUESTED** [review]

File: db/src/lookup_list.rs (hunk 2 above)
Comment (ad-tra) [reply db/src/lookup_list.rs]: it's required for `lookup_set::LogEntry<K, V>` to implement `Deserialize<'_>`. it doesn't compile when you remove it.

**Review by Parth — APPROVED** [review]

**Pull request merged**